### PR TITLE
perf: use react query in bridge tabs

### PR DESCRIPTION
**Pull request by flagrede**
## Description

Closes: regen-network/regen-registry#1464
- refactor [useBridged](https://github.com/regen-network/regen-web/pull/1771/files#diff-667ba8b62441329d374810f296bb703ff02490065652ceacd4141f436b5fb744) and [getBridgedEcocreditsForAccount](https://github.com/regen-network/regen-web/pull/1771/files#diff-04bbe8e9d39037506257b45de9ca801a42f31ca8d868e0421b78dc0a2dd2b7e2L216-L282) to [useFetchBridgedEcocredits](https://github.com/regen-network/regen-web/pull/1771/files#diff-e1affe9d17f850340d0b66b414f0fa7717967212aea590c6ba9648a90969468b)
- replace [useBridegeable](https://github.com/regen-network/regen-web/pull/1771/files#diff-3dc5a828ac99557647daabfcd567826ba5aa8a34b25b402052ed5efd1483baec) with [useFetchEcocredits](https://github.com/regen-network/regen-web/pull/1771/files#diff-3c6eb0830ad0060800e8963a3b13df56d6910bbe8569388bde67ad0006b613cb)
- [paginate](https://github.com/regen-network/regen-web/pull/1771/files#diff-e1affe9d17f850340d0b66b414f0fa7717967212aea590c6ba9648a90969468bR95) BridgeTxStatus queries
- add [polling](https://github.com/regen-network/regen-web/pull/1771/files#diff-e1affe9d17f850340d0b66b414f0fa7717967212aea590c6ba9648a90969468bR97-R104) to pending status along with a [loader and a tooltip](https://github.com/regen-network/regen-web/pull/1771/files#diff-f1f21d0fe8e9869fd0076ecdcd2af7eedafeb9a1a1465c958c53ba27c5885988R88-R104)

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] provided a link to the relevant issue or specification
- [ ] provided instructions on how to test
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed
- [ ] once the PR is closed, set up backport PRs for `redwood`, `hambach` and optionally `master` (see below)

#### Setting up backport PRs

After merging your PR to `dev`, set up backports by doing the following:

1. If your branch does not have merge commits, add the following comment to
   your PR:

- If PR is a new feature: `@Mergifyio backport redwood hambach`.
- If PR is a bug fix: `@Mergifyio backport redwood hambach master`.

2. If your branch does have merge commits:

   a. Pull latest `dev`, `hambach` and `redwood` branches

   b. Create new branches for backports and merge `dev` (replace `<PR#>` with your PR #)

   ```
   git checkout -b hambach-backport-<PR#> hambach
   git merge dev
   git push origin hambach-backport-<PR#>
   git checkout -b redwood-backport-<PR#> redwood
   git merge dev
   git push origin redwood-backport-<PR#>`
   ```

   c. Open new PRs in regen-web targeting `hambach` and `redwood`, respectively.

   d. If PR is a bug fix repeat the same process for `master` branch.

### How to test

1. Go to https://deploy-preview-1771--regen-registry.netlify.app/ecocredits/bridge
2. Use the shared test account `regen1df675r9vnf7pdedn4sf26svdsem3ugavgxmy46`
3. Bridge a small amount (ie: `0.001`)
4. Check the `Bridged Credits` tab right after
5. The new tx should appear in pending status and refresh once every 10 seconds

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] verified React components follow DRY principles
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

### CODE REVIEW

**Comment by flagrede** [comment]
@erikalogie you can test the improvements on the bridge tabs (polling + pagination on tx status), see the test section in the description. Note that the polling UX I've made is a proposition (loader + tooltip in header column), let me know what you think of it.

File: web-registry/src/components/organisms/BridgedEcocreditsTable/BridgedEcocreditsTable.constants.ts [hunk 1]
```diff
@@ -1,11 +1,15 @@
...
 export const STATUS_ERROR = 'error';
+export const TX_STATUS_REFRESH_INTERVAL = 10000;
+export const ONE_MINUTE_MS = 60000;
+export const TWO_MINUTE_MS = ONE_MINUTE_MS * 2;
 
 export const BRIDGED_STATUSES = {
   regen_hash_not_found: STATUS_PENDING,
   regen_ready: STATUS_PENDING,
   evm_broadcast: STATUS_PENDING,
+  evm_have_height: STATUS_PENDING,
```
Comment (flagrede) [inline web-registry/src/components/organisms/BridgedEcocreditsTable/BridgedEcocreditsTable.constants.ts:12]: We were missing this status before.

File: web-registry/src/pages/Dashboard/MyEcocredits/hooks/useFetchEcocredits.ts [hunk 2]
```diff
@@ -27,9 +28,15 @@ interface Response {
 
 interface Props {
   address?: string;
+  creditClassId?: string;
```
Comment (flagrede) [inline web-registry/src/pages/Dashboard/MyEcocredits/hooks/useFetchEcocredits.ts:31]: Updated the hook to be used with a `creditClassId`filter and without pagination.

File: web-registry/src/components/organisms/BridgedEcocreditsTable/BridgedEcocreditsTable.constants.ts (hunk 1 above)
Comment (blushi) [reply web-registry/src/components/organisms/BridgedEcocreditsTable/BridgedEcocreditsTable.constants.ts:12]: good catch, just got added a few days ago actually https://github.com/rnd-inc/toucan-bridge-service/pull/303

**Comment by erikalogie** [comment]
> @erikalogie you can test the improvements on the bridge tabs (polling + pagination on tx status), see the test section in the description. Note that the polling UX I've made is a proposition (loader + tooltip in header column), let me know what you think of it.

This looks awesome! What do you think about moving the little loader gif thingy into the "pending" pill itself (replacing the current icon?) https://monosnap.com/file/tuxhFgqiJ0s3Xu3TzzLJMKZMGNmmbP. I love the tooltip, let's keep that for sure!

**Comment by flagrede** [comment]
hey @erikalogie , so I gave a try to your suggestion, I was able to put the loader there but the problem was that the HTTP call is so fast that it's not even displayed. In the header column, I used a tricked to make it last longer than I cannot reuse easily with a loader inside each cell. So maybe let's keep it that way for now if it's okay for you.

**Comment by erikalogie** [comment]
> hey @erikalogie , so I gave a try to your suggestion, I was able to put the loader there but the problem was that the HTTP call is so fast that it's not even displayed. In the header column, I used a tricked to make it last longer than I cannot reuse easily with a loader inside each cell. So maybe let's keep it that way for now if it's okay for you.

Ok sure, sounds good!

**Review by wgwz — APPROVED** [review]
utACK

**Comment by flagrede** [comment]
@Mergifyio backport redwood hambach

**Pull request merged**